themes/opentermsarchive/layouts/partials/contributors.html

```diff
@@ -1,30 +1,51 @@
-{{ $thumbnailWidth := 64 }}
-{{ if .thumbnailWidth }}{{ $thumbnailWidth = .thumbnailWidth }}{{ end }}
-{{ $thumbnailHeight := 64 }}
-{{ if .thumbnailHeight }}{{ $thumbnailHeight = .thumbnailHeight }}{{ end }}
+{{ $thumbnailWidth := or .thumbnailWidth 64 }}
+{{ $thumbnailHeight := or .thumbnailHeight 64 }}
 
-{{ $allcontributorsrc := resources.Get "allcontributorsrc.json" | unmarshal }}
-{{ $filteredContributors := slice }}
+{{ $contributors := (resources.Get "allcontributorsrc.json" | unmarshal).contributors }}
 {{ $allStaffNames := slice }}
 {{ range site.Data.staff }}{{ $allStaffNames = union $allStaffNames . }}{{ end }}
 
-{{ range $allcontributorsrc.contributors }}
-  {{ $contributor := . }}
+{{ $remoteContributors := slice }}
+{{ with resources.GetRemote "https://raw.githubusercontent.com/OpenTermsArchive/engine/main/.all-contributorsrc" }}
+  {{ with .Err }}
+    {{ errorf "%s" . }}
+  {{ else }}
+    {{ $remoteContributors = (.Content | unmarshal).contributors }}
+  {{ end }}
+{{ else }}
+  {{ errorf "Unable to get the remote contributors" }}
+{{ end }}
+
+{{ $uniqueContributors := $contributors }}
+{{ range $remoteContributor := $remoteContributors }}
```

Comment:
The code in this loop seems not efficient and verbose.

Wouldn't it be better to use the `merge` function? Something like this (not sure about the code):
```go
{{ $contributorMap := dict }}
{{ range $contributors }}
  {{ $contributorMap = $contributorMap | merge (dict .name .) }}
{{ end }}

{{ range $remoteContributor := $remoteContributors }}
  {{ if not (index $contributorMap $remoteContributor.name) }}
    {{ $contributors = $contributors | append $remoteContributor }}
  {{ end }}
{{ end }}

{{ $uniqueContributors := $contributors }}
```

Reply:
You're absolutely right, I'd tried to improve but I hadn't figured out how to do it. Your feedback has helped me by trying again, which enabled me to find a solution 👍